api/db/migrations/20230928084925_flag_certifications_before_july_21_OUTDATED.js

```diff
@@ -0,0 +1,19 @@
+const TABLE_NAME = 'certification-courses';
+const COLUMN_NAME = 'cpfImportStatus';
+let certificationCourseIds;
+
+const up = async function (knex) {
+  certificationCourseIds = await knex(TABLE_NAME).select('id').where('createdAt', '<', '2021-07-01').pluck('id');
```

Comment:
Ca devrait passer mais si on a BEAUCOUP d'ids, on peut tomber dans un défaut de RAM. 
Option 1:
```suggestion
await knex(TABLE_NAME)
    .update({ [COLUMN_NAME]: 'OUTDATED' })
    .whereIn('id', knex(TABLE_NAME).select('id').where('createdAt', '<', '2021-07-01').pluck('id'));
```
option 2: 
utiliser un `with`

Si vraiment gros probleme de perf, on pourrait aussi faire en fonction de l'id et pas la date (mais en fonction de l'environnement ca ne fonctionnerait pas pareil)

Reply:
@aceol  Avec l'option 1 on fait tout en une requête, on ne risque pas de péter encore plus la ram ?

Reply:
@P-Jeremy Dans ton option c'est ton backend, ton API qui porte la charge d'avoir les ID dans sa mémoire (+ l'aller-retour réseau et les risques à passer sur un network).

Dans l'option 1 de @aceol c'est la database qui portera la charge, car le SELECT des ids devient une SUBQUERY SQL.

La version de @aceol est donc préférable.

Reply:
Pour pas que ça reste que à l'oral sur l'option 2.
Je proposais de passer du `WHERE IN` à un `WHERE EXISTS` et c'est là où le `with` de @aceol devient intéressant.

```sql
WITH(cte-outaded-table, select ... 'createdAt', '<', '2021-07-01')
UPDATE ... as myTable
WHERE id EXISTS (cte-outaded-table.id = myTable.id)
```

Avec la chose importante qui est la suivante :

> A noter : cette commande n’est pas à confondre avec la clause IN. La commande EXISTS vérifie si la sous-requête retourne un résultat ou non, tandis que IN vérifie la concordance d’une à plusieurs données.

Et pour détailler pourquoi le EXISTS est intéressant ici, quand une subquery renvoie potentiellement beaucoup de lignes (ça sera le cas en PROD), alors EXISTS est utile car il s'arrête à la première occurence (oui/non) ce qui n'est pas le cas de IN (1 à N, mais il faut qu'il trouve les N, il s'arrête pas à la première).